frontend/src/app/superadmin/users/users.component.ts

```diff
@@ -216,10 +216,20 @@ export class UsersComponent implements OnInit {
   }
 
   selectWorkspace(ws: IdRoleData, role: string): void {
```

Comment:
Insgesamt ist diese Methode sehr komisch. Sie heisst "selectWorkspaces". In der Methode scheints aber nur um Berechtigungen zu gehen...
Außerdem wird der Parameter mit Typ IDRoleData mit "ws" bezeichnet.
Finde es außerdem komisch, dass die role in IDRoleData in einem Fall mit einem leeren String belegt wird.

Und allem die Krone setzt die komplett unklare Logik in der Methode auf. Keine Ahnung, ob du die durchdrungen hast, mir fällts schwer. Ich habe auch den Verdacht, dass man das alles kurzer und prägnanter hinschreiben könnte.

Müsste eigentlich alles mal bereinigt werden. Vermutlich am besten einmal großflächig niederbrennen und ordentlich neu machen. Aber das wäre hier zuviel verlangt... :D